internal/traceinternals.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import "time"
+
+// Trace allows internal access to some trace functionality.
+// TODO(#412): remove this
+var Trace interface{}
```

Comment:
var Reporter interface {
 	ReportActiveSpans(name string) []*trace.SpanData
 	ReportSpansByError(name string, code int32) []*trace.SpanData
 	ReportSpansByLatency(name string, minLatency, maxLatency time.Duration) []*trace.SpanData
        ReportSpansPerMethod() map[string]internal.PerMethodSummary
}

Reply:
reason it's untyped is to avoid a circular dependency

Reply:
Ah sorry, cool trick :)